float32[12] states
